src/libpmemobj/tx.c

```diff
@@ -1124,23 +1282,44 @@ pmemobj_tx_begin(PMEMobjpool *pop, jmp_buf env, ...)
 	LOG(3, NULL);
 
 	int err = 0;
-	struct tx *tx = get_tx();
+	struct tx *tx = tx_ctx_get();
+
+	if (pop == NULL) {
```

Comment:
I don't get this. Why would you ever call pmemobj_tx_begin() with a NULL pop?
I'm pretty sure this would just crash previously, and we should maintain that behavior (maybe add an ASSERT).